### PR TITLE
Build Apps using custom docker images

### LINKED ISSUE #184: Functionality for running custom docker images supplied by users

### PR DESCRIPTION
**Pull request by sukhman-sukh**

### CODE REVIEW

File: lib/api/application.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ func setupContainer(app types.Application, storedir string, setup chan types.Res
 		Name:            app.GetName(),
 		Image:           app.GetDockerImage(),
 		ApplicationPort: app.GetApplicationPort(),
-		ContainerPort:   app.GetContainerPort(),
+		ContainerPort:   app.GetContainerPort(),
```
Comment (itsdarshankumar) [inline lib/api/application.go]: remove extra space

File: lib/docker/image.go [hunk 2]
```diff
@@ -4,11 +4,32 @@ import (
...
 	"os/exec"
+	"strings"
 
 	"github.com/docker/docker/api/types"
+	"github.com/sdslabs/gasper/lib/utils"
 	"golang.org/x/net/context"
 )
 
+// Check for available images and pull if not present
+func CheckAndPullImages(imageList ...string) {
```
Comment (itsdarshankumar) [inline lib/docker/image.go:15]: any reason why to migrate this here from helper?

**Review by itsdarshankumar — CHANGES_REQUESTED** [review]

File: lib/api/application.go (hunk 1 above)
Comment (sukhman-sukh) [reply lib/api/application.go]: I think ye formatter ne daal diya hoga,
I will remove it

File: lib/docker/image.go (hunk 2 above)
Comment (sukhman-sukh) [reply lib/docker/image.go:15]: Yes,
1. This function is specific to docker so it should be in docker package only, I think.
2. I had to reuse the above function and can't import functions from global files but packages. So, I thought better to migrate.

File: lib/api/application.go (hunk 1 above)
Comment (itsdarshankumar) [reply lib/api/application.go]: iska commit daalde baaki LGTM

Comment (sukhman-sukh) [reply lib/api/application.go]: done

**Pull request merged**